build_projects/dotnet-host-build/build.ps1

```diff
@@ -26,9 +32,39 @@ if($Help)
 }
 
 $env:CONFIGURATION = $Configuration;
+if (!$TargetArch)
+{
+    $TargetArch = $Architecture
+}
+$env:TARGETPLATFORM = $TargetArch;
+$env:TARGETFRAMEWORK = $Framework;
 $RepoRoot = "$PSScriptRoot\..\.."
 $env:NUGET_PACKAGES = "$RepoRoot\.nuget\packages"
 
+if($TargetArch -eq "arm64")
+{
+    if ($Framework -eq "netcoreapp1.0")
+    {
+        throw "ARM64 is not available on netcoreapp1.0. Pass in '-Framework netcoreapp1.1' to enable ARM64"
+    }
+    $env:__ToolsetDir = $ToolsetDir;
+    $env:TARGETRID = "win10-arm64";
+    $env:DISABLE_CROSSGEN = "1";
+    $env:PATH="$ToolsetDir\VC_sdk\bin;$env:PATH";
+    $env:LIB="$ToolsetDir\VC_sdk\lib\arm64;$ToolsetDir\sdpublic\sdk\lib\arm64";
+    $env:INCLUDE="$ToolsetDir\VC_sdk\inc;$ToolsetDir\sdpublic\sdk\inc;$ToolsetDir\sdpublic\shared\inc;$ToolsetDir\sdpublic\shared\inc\minwin;$ToolsetDir\sdpublic\sdk\inc\ucrt;$ToolsetDir\sdpublic\sdk\inc\minwin;$ToolsetDir\sdpublic\sdk\inc\mincore;$ToolsetDir\sdpublic\sdk\inc\abi;$ToolsetDir\sdpublic\sdk\inc\clientcore;$ToolsetDir\diasdk\include";
+}
+
+# No use in specifying a RID if the current and target architecture are equivalent. 
+if($TargetArch -eq "x86" -and $Architecture -ne "x86")
```

Comment:
> Do the artifacts go in setup/artifacts/win10-x64 like they used to or should they go in setup/artifacts/win7-x64 since that's the true TargetRID?

I believe the RID's premise is to assume that what you are building upon is what you are going to target as the platform - unless you have been specified the target RID (e.g. in cross compilation scenarios like arm32 or arm64). I believe you should keep the existing behaviour but enable it by specifying the RID to be used:

1) For targeting scenarios, it will be the RID computed by default
2) For cross-targeting scenarios, it will be the RID calculated.
3) When specified RIDs, it will go into the intended RID folder